opm/grid/CpGrid.hpp

```diff
@@ -66,7 +66,7 @@ class NNC;
 
 namespace Dune
 {
-
+    template<typename Grid> class CartesianIndexMapper;
```

Comment:
I think with we don't need this and the friend declaration anymore because of `chooseData`.